src/vnsw/opencontrail-vrouter-netns/opencontrail_vrouter_netns/vrouter_hyperv.py

```diff
@@ -0,0 +1,469 @@
+        create_parser.add_argument(
+            "vm_id",
+            help="Virtual machine UUID")
+        create_parser.add_argument(
+            "vmi_left_id",
+            help="Left virtual machine interface UUID")
+        create_parser.add_argument(
+            "vmi_right_id",
+            help="Right virtual machine interface UUID")
+        create_parser.add_argument(
+            "--vmi-left-mac",
+            default=None,
+            help=("Left virtual machine interface MAC. Default: automatically "
+                  "generated by the system"))
+        create_parser.add_argument(
+            "--vmi-left-ip",
+            default=None,
+            help=("Left virtual machine interface IPv4 and mask "
+                  "(ie: a.a.a.a/bb). Default mask to /32"))
+        create_parser.add_argument(
+            "--vmi-right-mac",
+            default=None,
+            help=("Right virtual machine interface MAC. Default: "
+                  "automatically generated by the system"))
+        create_parser.add_argument(
+            "--vmi-right-ip",
+            default=None,
+            help=("Right virtual machine interface IPv4 and mask "
+                  "(ie: a.a.a.a/bb). Default mask to /32"))
+        create_parser.add_argument(
+            "--gw-ip",
+            default=None,
+            help=("Gateway IP for Virtual Network"))
+        create_parser.set_defaults(func=self.create)
+
+        destroy_parser = subparsers.add_parser('destroy')
+        destroy_parser.add_argument(
+            "vm_id",
+            help="Virtual machine UUID")
+        destroy_parser.add_argument(
+            "vmi_left_id",
+            help="Left virtual machine interface UUID")
+        destroy_parser.add_argument(
+            "vmi_right_id",
+            help="Right virtual machine interface UUID")
+        destroy_parser.set_defaults(func=self.destroy)
+
+        self.args = parser.parse_args(remaining_argv)
+
+    def create(self):
+        vm_id = validate_uuid(self.args.vm_id)
+
+        nic_left = {}
+        if uuid.UUID(self.args.vmi_left_id):
+            nic_left['uuid'] = validate_uuid(self.args.vmi_left_id)
+            if self.args.vmi_left_mac:
+                nic_left['mac'] = netaddr.EUI(self.args.vmi_left_mac,
+                                              dialect=netaddr.mac_eui48) # does eui48 break sth?
+            else:
+                nic_left['mac'] = None
+            if self.args.vmi_left_ip:
+                nic_left['ip'] = netaddr.IPNetwork(self.args.vmi_left_ip)
+            else:
+                nic_left['ip'] = None
+
+        nic_right = {}
+        if uuid.UUID(self.args.vmi_right_id):
+            nic_right['uuid'] = validate_uuid(self.args.vmi_right_id)
+            if self.args.vmi_right_mac:
+                nic_right['mac'] = netaddr.EUI(self.args.vmi_right_mac,
+                                               dialect=netaddr.mac_eui48) # does eui48 break sth?
```

Comment:
Using `mac_eui48` as dialect will produce MAC address in following format: `DE-AD-BE-EF-00-00`, which is the standard Windows format. So probably when ports are registered in Contrail we should convert it to format used by Linux, i.e. with colons.